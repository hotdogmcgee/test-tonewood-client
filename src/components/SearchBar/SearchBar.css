#SearchBar__Section {
  display: flex;
  flex-direction: row;
  justify-content: center;
  margin: 2rem 0;
}

#SearchForm__entry {
  text-align: left;
  width: 200px;
  margin-left: 15px;
}

#SearchForm__entry::placeholder {
  opacity: 0.5;
}

@media screen and (min-width: 800px) {
  #SearchBar__Section {
    justify-content: flex-start;
    padding-left: 20px;
  }
}
