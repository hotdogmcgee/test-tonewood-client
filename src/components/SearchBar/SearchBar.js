import React from "react";
import { Input } from "../Utils/Utils";

export default class SearchBar extends React.Component {

static defaultProps = {
    onSearchChange: () => {}
}

handleInput = e => {
    e.preventDefault()
    const searchValue = e.target.value
    this.props.onSearchChange(searchValue)
}
  render() {
    return (
      <form className="SearchForm" onChange={this.handleInput}>
        <label htmlFor="search-bar">Search</label>
        <Input placeholder="tonewood" name="search-bar" id='SearchForm__entry'></Input>
      </form>

    );
  }
}
