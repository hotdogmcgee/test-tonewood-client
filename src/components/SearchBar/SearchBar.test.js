import React from "react";
import SearchBar from "./SearchBar";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("SearchBar component", () => {
  it("renders a SearchBar by default", () => {
    const wrapper = shallow(<SearchBar />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});