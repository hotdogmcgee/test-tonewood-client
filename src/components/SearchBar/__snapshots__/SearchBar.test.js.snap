// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component renders a SearchBar by default 1`] = `
<div
  autoComplete="off"
  className="SearchForm"
  onChange={[Function]}
>
  <Input
    autoComplete="off"
    id="SearchForm__entry"
    name="search-bar"
    placeholder="search"
  />
</div>
`;
